.github/workflows/main.yml

```diff
@@ -1,53 +1,80 @@
+# actions can be run locally using act and docker, on Fedora 37 also with podman, using:
+# https://github.com/nektos/act
+# sudo dnf install -y act-cli podman-docker
+# act --bind --container-daemon-socket $XDG_RUNTIME_DIR/podman/podman.sock -W .github/workflows/main.yml
+
 name: Unit tests
 
+# Checks can be skipped by adding "skip-checks: true" to a commit message,
+# or requested by adding "request-checks: true" if disabled by default for pushes:
+# https://docs.github.com/en/pull-requests/collaborating-with-pull-requests/collaborating-on-repositories-with-code-quality-features/about-status-checks#skipping-and-requesting-checks-for-individual-commits
 on: [push, pull_request]
+env:
+  PYTHONWARNINGS: "ignore"
+  PIP_ROOT_USER_ACTION: "ignore"   # For local testing using act-cli
+  PIP_NO_WARN_SCRIPT_LOCATION: "0" # For local testing using act-cli
 
 jobs:
   test:
     strategy:
+      # max-parallel: 1  # See: https://github.com/xenserver/python-libs/pull/26#discussion_r1179482169
       matrix:
         include:
+        - python-version: '3.11'
+          os: ubuntu-latest
+        - python-version: '3.10'
+          os: ubuntu-latest
         - python-version: '2.7'
           os: ubuntu-20.04
-        - python-version: '3'
+        - python-version: '3.9'
+          os: ubuntu-latest
+        - python-version: '3.8'
           os: ubuntu-latest
+        - python-version: '3.7'
+          os: ubuntu-latest
+        - python-version: '3.6'
+          os: ubuntu-20.04
     runs-on: ${{ matrix.os }}
 
     steps:
       - uses: actions/checkout@v3
         with:
-          fetch-depth: 0
+          fetch-depth: 0  # Needed by diff-cover to get the changed lines: origin/master..HEAD
       - name: Set up Python ${{ matrix.python-version }}
         uses: actions/setup-python@v4
         with:
           python-version: ${{ matrix.python-version }}
 
-      - name: Install dependencies
+      - name: Provide Pylint Summary and Report on the GitHub Action's info page
+        if: ${{ matrix.python-version == '3.10' }}
         run: |
-          python -m pip install --upgrade pip
-          pip install -r requirements-dev.txt
-          pip install -e .
+          pip install -q pylint pandas tabulate 2>&1|grep -v root ||: # Hide warning for root in local container
+          tree=$GITHUB_SERVER_URL/$GITHUB_REPOSITORY/tree/$GITHUB_REF_NAME
+          [ -n "$GITHUB_STEP_SUMMARY" ] || GITHUB_STEP_SUMMARY=pylint-summary-table.md
+          ./gen_gh_pylint_report.py xcp $GITHUB_STEP_SUMMARY $tree
 
       - name: Test
         run: |
-          pytest --cov -rP
-          coverage xml
-          coverage html -d htmlcov-${{ matrix.python-version }}
-          coverage html -d htmlcov-tests-${{ matrix.python-version }} --include="tests/*"
-          diff-cover --compare-branch=origin/master --html-report coverage-diff-${{ matrix.python-version }}.html coverage.xml
+          rm -f coverage.xml
```

Comment:
Not sure why this removal is needed, every CI run is usually run on a clean slate

Reply:
Thanks, removed in PR 27!